tripy/tripy/frontend/module/sequential.py

```diff
@@ -0,0 +1,169 @@
+from typing import Union, Tuple, Dict, Iterator, Any
+from dataclasses import dataclass
+
+from tripy import export
+from tripy.frontend.module.parameter import Parameter
+from tripy.frontend.module import Module
+
+
+@export.public_api(document_under="modules/sequential.rst")
+@dataclass
+class Sequential(Module):
+    r"""
+    A module to stack multiple layers or modules in a sequential order. The `Sequential`
+    container can accept either a list of modules or a dictionary of named modules. Modules are
+    added in the order they are passed, and each is called sequentially during the forward pass.
+    """
+
+    modules: Union[Module, Dict[str, Module]]
+    r"""The modules to include in the sequence"""
+
+    def __init__(self, *modules: Union[Module, Dict[str, Module]]) -> None:
+        r"""
+        Args:
+            *modules (Union[Module, Dict[str, Module]]): The modules to include in the sequence.
+                Can be passed as individual positional arguments or as a single dictionary of named modules.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            import tripy as tp
+
+            # Sequential with layers passed as arguments
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+
+            # Sequential with named dictionary of layers
+            model_named = tp.Sequential({'layer1': tp.Linear(1, 3), 'layer2': tp.Linear(3, 2)})
+
+            x = tp.Tensor([1.0])
+            output = model(x)
+        """
+        super().__init__()
+        self.modules = {}
+
+        if len(modules) == 1 and isinstance(modules[0], dict):
```

Comment:
Should we disallow mixing modules/dicts in the constructor?

Reply:
done

Reply:
After thinking about this some more, I think it's probably better to let the function registry's type checking handle this so we don't need to duplicate type checking logic. 

For that to work we will need to implement #338 and #341 but I think we can omit this check for now.

Reply:
okay, sounds good. removing

Reply:
sounds good, removing!